custom-recipes/nlp-spell-checker/recipe.json

```diff
@@ -0,0 +1,157 @@
+             {'label': 'Bulgarian', 'value': 'bg'},
+             {'label': 'Catalan', 'value': 'ca'},
+             {'label': 'Chinese', 'value': 'zh'},
+             {'label': 'Croatian', 'value': 'hr'},
+             {'label': 'Czech', 'value': 'cs'},
+             {'label': 'Danish', 'value': 'da'},
+             {'label': 'Dutch', 'value': 'nl'},
+             {'label': 'English', 'value': 'en'},
+             {'label': 'Estonian', 'value': 'et'},
+             {'label': 'Finnish', 'value': 'fi'},
+             {'label': 'French', 'value': 'fr'},
+             {'label': 'German', 'value': 'de'},
+             {'label': 'Greek', 'value': 'el'},
+             {'label': 'Hebrew', 'value': 'he'},
+             {'label': 'Hungarian', 'value': 'hu'},
+             {'label': 'Icelandic', 'value': 'is'},
+             {'label': 'Indonesian', 'value': 'id'},
+             {'label': 'Italian', 'value': 'it'},
+             {'label': 'Japanese', 'value': 'ja'},
+             {'label': 'Korean', 'value': 'ko'},
+             {'label': 'Latvian', 'value': 'lv'},
+             {'label': 'Lithuanian', 'value': 'lt'},
+             {'label': 'Persian', 'value': 'fa'},
+             {'label': 'Polish', 'value': 'pl'},
+             {'label': 'Portuguese', 'value': 'pt'},
+             {'label': 'Romanian', 'value': 'ro'},
+             {'label': 'Russian', 'value': 'ru'},
+             {'label': 'Serbian', 'value': 'sr'},
+             {'label': 'Slovak', 'value': 'sk'},
+             {'label': 'Slovenian', 'value': 'sl'},
+             {'label': 'Spanish', 'value': 'es'},
+             {'label': 'Swedish', 'value': 'sv'},
+             {'label': 'Thai', 'value': 'th'},
+             {'label': 'Turkish', 'value': 'tr'},
+             {'label': 'Ukrainian', 'value': 'uk'},
+             {'label': 'Vietnamese', 'value': 'vi'}],
+            "defaultValue": "en",
+        "visibilityCondition": "model.language_selection == 'from_list'"
+        },
+        
+        {
+            "name": "language_from_column",
+            "label": "Language column",
+            "type": "COLUMN",
+            "description": "Column containing language ISO 639-1 codes",
+            "columnRole": "input_dataset",
+            "allowedColumnTypes": [
+                "string"
+            ],
+            "visibilityCondition": "model.language_selection == 'from_column'"
+
+        },
+        {
+        "name": "separator_advanced",
+        "label": "Advanced",
+        "type": "SEPARATOR"
+
+          },
+        {
+            "name": "expert",
+            "label": "Expert mode",
+            "type": "BOOLEAN",
+            "defaultValue": false
+        },
+        {
+        "type": "STRING",
+        "name": "ignore_token",
+        "label": "Ignore token",        
+        "defaultValue": "",
+        "description": "Regular expression pattern for words/phrases not to be corrected. Ex: \\w+\\d will preserve \"home1\" ",
```

Comment:
hashtags and mentions are removed by the tokenizer. So they are lost using symspell.

Reply:
I understand that, but the \w+ example isn't the most appropriate in my opinion. In the context of spelling correction, you would actually not use \w+ as an ignore pattern. That would mean no words are corrected. I think we should find a better example, which would be valid in the context of spelling correction.

Reply:
What about "containing both letters and number", as an ID would do ?

Reply:
Yes, that would be clearer. I would also add somewhere that hashtags and mentions are directly removed, so that the user knows he doesn't need to take care of that.

Reply:
We should handle this more carefully. 
-> Hashtags and mentions can be great in a NLP pipeline: wordcloud, classification, ...
-> keeping them and not treating would make the code slower
-> not doing anything would lead to correct the hashtags into words of the dictionary
Any thoughts on this ?

Reply:
You raise a good point. For spell checker is OK to remove hashtags and mentions. But for other plugins, it's going to be a problem. 
Is it possible to prevent the tokenizer from doing this filtering by default? 
And then in that case, we can use the ignore_token parameter as intended, to remove them?

Reply:
Yes, we should do this. I put it on my to do.
In case it adds a lot of computation time, we could set this as an option.

Reply:
Good idea